src/file/writer.js

```diff
@@ -188,11 +197,23 @@ export const link = (state, { id, link, block }) => {
       ? state.end.resume()
       : Task.none()
 
+  if (!state.unixFsFileLinkWriter) {
+    return {
+      state: newState,
+      effect: Task.listen({
+        link: Task.effects(tasks),
+        block: writeBlock(state.writer, block),
+        end,
+      }),
+    }
+  }
+
   return {
     state: newState,
     effect: Task.listen({
       link: Task.effects(tasks),
       block: writeBlock(state.writer, block),
+      fileLink: writeFileLink(state.unixFsFileLinkWriter, link),
```

Comment:
```suggestion
  return {
    state: newState,
    effect: Task.listen({
      link: Task.effects(tasks),
      block: writeBlock(state.writer, block),
      fileLink: state.unixFsFileLinkWriter ? writeFileLink(state.unixFsFileLinkWriter, link) : undefined,
```

Nit: This feels more straightforward to me personally.

P.S.: See my other more comments that make this remark obsolete